hbase-common/src/main/java/org/apache/hadoop/hbase/KeyValue.java

```diff
@@ -230,6 +230,11 @@ public static long getKeyDataStructureSize(int rlength, int flength, int qlength
     DeleteColumn((byte)12),
     DeleteFamily((byte)14),
 
+    // Effective maximum is 127 (Byte.MAX_VALUE). We set the high order bit of the
+    // type byte in the WAL codecs to indicate, in a backwards compatible way, if the
+    // value is compressed there.
+    EffectiveMaximum((byte)Byte.MAX_VALUE),
+
     // Maximum is used when searching; you look from maximum on down.
     Maximum((byte)255);
```

Comment:
> Add explicit Type entries for CompressedPut, CompressedDelete, &c., with their associated code values also explicitly listed

Such Type entries would not be used anywhere outside of the WAL codec. It's really a WAL codec internal detail. As such, this would raise more questions than provide answers. However I agree with your related point that to make this code production ready the magic constants should be defined with names and javadoc (in wal codec)

Reply:
It is simpler if we don't try to conditionally compress values. No high bit twiddling. So there is no leakage into KeyValue in this case and no magic values.